### ISSUE TITLE
[Feature]  properly support <details>

### ISSUE THREAD

**Issue by Pomax** [typora/typora-issues#5959]
1.8.6 came with updates for GitHub flavoured markdown, but ever since Typora got released, the `<details>` solution for collapsible sections [as officially advocated by github themselves](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections) hasn't been (properly) supported. 

The linked document shows exactly how it's supposed to work, so... can we please get Typora to do what GitHub does, when writing GFM, ideally in the next minor release?

(Even if  you don't add full HTML support, which ideally you should, because that's officially [part of GFM](https://github.github.com/gfm/#html-blocks), at least add proper support for the details element)

**Comment by Pomax** [comment]
To wit, using the official GitHub specified example code:

````markdown
<details>

<summary>Tips for collapsed sections</summary>

### You can add a header

You can add text within a collapsed section. 

You can add an image or a code block, too.

```ruby
   puts "Hello World"
```

</details>
````

Typora renders this:

![image](https://github.com/typora/typora-issues/assets/177243/18892994-f1cf-4559-a5f2-0e798ec49c03)

So it looks like it has some built in CSS to add the little triangle, but that's about all the support it currently has.

**Comment by hsandt** [comment]
Indeed, the fact that the details section is broken and the closing `</details>` pops up at the end also has unintended consequences, such as copying HTML trying to wrap `</details>` into its own `<p>...</p>` tags.

I use this to copy folded sections into itch.io game descriptions directly as HTML (due to the lack of Markdown editor for game descriptions), and it causes a visible `</p>` to appear after the folded section. I have to manually move the `</p>` back before `</details>`.

**Comment by hsandt** [comment]
I found the exact conditions to trigger `</details>` bad `<p>` wrapping and more, and since I described a different issue related to Copy as HTML and not display in Typora, I opened a new ticket: https://github.com/typora/typora-issues/issues/6414

That said, I found out that blank lines were _also_ responsible for the display issue mentioned by the OP.

So this:

```md
<details>
<summary>Tips for collapsed sections</summary>
### You can add a header
You can add text within a collapsed section. 
You can add an image or a code block, too.
```ruby
   puts "Hello World"
```
</details>
```

will display correctly (and also copy as HTML correctly).

But that's less readable. Obviously you can add `<br>` for an _actual_ line break, but sometimes you just want to make things clearer for the editor (not reader) by adding blank lines.

To summarize, the issue behind both `<details>` display and Copy as HTML is that Typora doesn't support blank lines inside `<details>` block.

Maybe this issue could be renamed "[Bug] Adding blank lines inside `<details>` block breaks details block parsing (breaks display and Copy as HTML)" and my issue would become a sub-issue of yours?

**Comment by emiliom** [comment]
Just to add to this, I'd like my `<details>` block to contain only a markdown fenced code block (lines surrounded by three back ticks, ``` ). If I remove all blank lines, including blank lines before and after the fenced code block, the `<details>` block behaves as expected: it can be collapsed and opened. However, that breaks the markdown rendering within the fenced code block, such that it becomes a long, continuous -- and effectively unreadable -- block of text. 

In fact, that seems to be the behavior for any markdown code within a `<details>` block where all blank lines are removed. To build on the example by the previous poster, but using screenshots for clarity, this:

<img width="290" height="203" alt="Image" src="https://github.com/user-attachments/assets/a5508338-5bc6-4a4a-a0f0-af7862e20300" />

is rendered like this:

<img width="556" height="76" alt="Image" src="https://github.com/user-attachments/assets/f0b7eeb7-1eba-4ba5-a2e2-a82ce81fad47" />


xref: #6408

**Comment by emiliom** [comment]
As of vers. 1.13.6, this still isn't fixed. But in case it helps anyone, I figured out I can meet most of my needs by following a couple of tricks. First, my typical need for a `<details>` block is to include a long text log (error reports, run-time logs, etc) using a markdown fenced code block (lines surrounded by three back ticks, ``` ). If I remove all blank lines, the `<details>` content will be hidden. Then, instead of opening the block by clicking on it (which will reveal an unusable, continuous blob of text), I *move* the cursor *into* the block by using the arrow keys. This will open the \<details\> block in "edit" mode. The content inside the fenced code block will be visible, with line breaks preserved, though code block formatters (python, sql, bash, etc) are not respected, so no syntax highlighting, no markdown formatting. Here's what that looks like:

<img width="399" height="209" alt="Image" src="https://github.com/user-attachments/assets/bc403c99-344b-48e1-acbe-5fb9ea30d4a8" />

The block closes when I click outside of it. Also, toggling to Source Code Mode will reveal the block with markdown formatting, if I need it.

 For most of my needs, that's workable. I get a collapsible block that I can open easily. A real fix would be really helpful, though!
